### PR TITLE
Adds --user and --devTrustStore flags to the cli

### PR DESCRIPTION
**Pull request by alokmenghrajani**
This should address https://github.com/square/keywhiz/issues/42, https://github.com/square/keywhiz/issues/64, and https://github.com/square/keywhiz/issues/65

### CODE REVIEW

File: cli/src/main/java/keywhiz/cli/CommandExecutor.java [hunk 1]
```diff
@@ -102,9 +102,17 @@ public void executeCommand() throws IOException {
     KeywhizClient client;
     OkHttpClient encapsulatedClient;
 
+    String user;
+    if ((config.user != null) && !config.user.isEmpty()) {
```
Comment (sul3n3t) [inline cli/src/main/java/keywhiz/cli/CommandExecutor.java]: `if (Strings.nullToEmpty(config.user).isEmpty())`

File: cli/src/main/java/keywhiz/cli/CommandExecutor.java [hunk 2]
```diff
@@ -102,9 +102,17 @@ public void executeCommand() throws IOException {
...
+      user = config.user;
+    } else {
+      user = USER_NAME.value();
+    }
+    Path cookiePath = cookieDir.resolve(".keywhiz." + user + ".cookie");
```
Comment (sul3n3t) [inline cli/src/main/java/keywhiz/cli/CommandExecutor.java]: Use `String.format()`

File: cli/src/main/java/keywhiz/cli/ClientUtils.java [hunk 3]
```diff
@@ -76,14 +79,20 @@
    * @param cookies list of cookies to include in the client.
    * @return new http client.
    */
-  public static OkHttpClient sslOkHttpClient(List<HttpCookie> cookies) {
+  public static OkHttpClient sslOkHttpClient(CliConfiguration config, List<HttpCookie> cookies) {
```
Comment (alokmenghrajani) [inline cli/src/main/java/keywhiz/cli/ClientUtils.java]: fixed this API.

File: cli/src/main/java/keywhiz/cli/CommandExecutor.java (hunk 2 above)
Comment (alokmenghrajani) [inline cli/src/main/java/keywhiz/cli/CommandExecutor.java]: Fixed

File: cli/src/main/java/keywhiz/cli/CommandExecutor.java [hunk 4]
```diff
@@ -51,13 +51,14 @@
...
 import static com.google.common.base.StandardSystemProperty.USER_NAME;
+import static java.lang.String.format;
 
 public class CommandExecutor {
   public static final String APP_VERSION = "2.0";
 
   public enum Command { LOGIN, LIST, DESCRIBE, ADD, DELETE, ASSIGN, UNASSIGN }
 
-  private final Path cookiePath = Paths.get(USER_HOME.value(), "/.keywhiz.cookies");
+  private final Path cookieDir = Paths.get(USER_HOME.value(), "/");
```
Comment (sul3n3t) [inline cli/src/main/java/keywhiz/cli/CommandExecutor.java]: Just `Paths.get(USER_HOME.value());` now

File: cli/src/main/java/keywhiz/cli/ClientUtils.java [hunk 5]
```diff
@@ -76,14 +76,17 @@
    * @param cookies list of cookies to include in the client.
```
Comment (sul3n3t) [inline cli/src/main/java/keywhiz/cli/ClientUtils.java]: Add a `@param` entry for the Keystore and explain what happens when null is supplied.

File: cli/src/main/java/keywhiz/cli/DevTrustStore.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+public class DevTrustStore {
+  private static final String store = "dev_and_test_truststore.p12";
+  private static final String password = "ponies";
+
+  public KeyStore getKeyStore() {
```
Comment (sul3n3t) [inline cli/src/main/java/keywhiz/cli/DevTrustStore.java]: javadoc, and maybe rename to `getTrustStore()`

File: cli/src/main/java/keywhiz/cli/ClientUtils.java (hunk 5 above)
Comment (alokmenghrajani) [inline cli/src/main/java/keywhiz/cli/ClientUtils.java]: done

File: cli/src/main/java/keywhiz/cli/CommandExecutor.java (hunk 4 above)
Comment (alokmenghrajani) [inline cli/src/main/java/keywhiz/cli/CommandExecutor.java]: fixed

File: cli/src/main/java/keywhiz/cli/DevTrustStore.java (hunk 6 above)
Comment (alokmenghrajani) [inline cli/src/main/java/keywhiz/cli/DevTrustStore.java]: done

**Comment by sul3n3t** [comment]
:+1:

**Pull request merged**